activemq-broker/src/main/java/org/apache/activemq/plugin/StatisticsBroker.java

```diff
@@ -160,6 +160,11 @@ else if (dest instanceof Topic) {
                                 tempFirstMessage.clear();
                             }
                         }
+                        if ( stats.getEnqueues().hasUpdated() ) {
+                            // NOTICE: Client-side, you may get the broker "now" Timestamp by msg.getJMSTimestamp()
+                            // This allows for calculating inactivity.
+                            statsMessage.setLong("lastMessageTimestamp", stats.getEnqueues().getLastSampleTime());
```

Comment:
@mattrpav I can see your change is merged, so I'm refreshing the pull request.

I understand the name unification motivation. Should I change name **lastMessageTimestamp** into **lastEnqueTimestamp** here?

Reply:
@ggkochanski The field naming is a broader conversation. The full scope is not one field as you have in this PR. We currently have a number of metrics (and config params) that are not fully aligned in naming convention, so I would press for an organized approach for these new metrics.

Related JIRAs:
ref: https://issues.apache.org/jira/browse/AMQ-4358
ref: https://issues.apache.org/jira/browse/AMQ-5497

Reply:
What is the meaning LastDequeueTime and FirstDequeueTime? I believe one of them cannot be determined (if **the last** means the newest enqueued message then it is not dequeued yet)

@mattrpav could you explain in details how you imagine this new approach to metrics based on some example preferably?

Maybe I try to use statistics in the wrong way. What I need to know to assess the condition of mq system is: enqueue count and last update timestamp, dequeue count and last update, last process time (iso "averageEnqueueTime") and the age of the oldest message in queue. What do you suggest me?

Reply:
@ggkochanski yeah, the challenge is settling on the terminology for the metric values. 

For example, does 'last' mean last-as-in-time, or last-as-in-last-message-in-the-queue (most recently added)